PrebidMobile/API1.0/src/main/java/org/prebid/mobile/NativeAdUnit.java

```diff
@@ -0,0 +1,371 @@
+ */
+public class NativeAdUnit extends AdUnit {
+    enum NATIVE_REQUEST_ASSET {
+        TITLE,
+        IMAGE,
+        DATA
+    }
+
+    // constants
+
+    static String VERSION = "ver";
+    static String SUPPORTED_VERSION = "1.2";
+    static String CONTEXT = "context";
+    static String CONTEXT_SUB_TYPE = "contextsubtype";
+    static String PLACEMENT_TYPE = "plcmttype";
+    static String PLACEMENT_COUNT = "plcmtcnt";
+    static String SEQ = "seq";
+    static String ASSETS = "assets";
+    static String A_URL_SUPPORT = "aurlsupport";
+    static String D_URL_SUPPORT = "durlsupport";
+    static String EVENT_TRACKERS = "eventtrackers";
+    static String PRIVACY = "privacy";
+    static String EXT = "ext";
+    static String EVENT = "event";
+    static String METHODS = "methods";
+    static String LENGTH = "len";
+    static String REQUIRED = "required";
+    static String ASSETS_EXT = "assetExt";
+    static String WIDTH_MIN = "wmin";
+    static String HEIGHT_MIN = "hmin";
+    static String WIDTH = "W";
+    static String HEIGHT = "h";
+    static String TYPE = "type";
+    static String MIMES = "mimes";
+    static String TITLE = "title";
+    static String IMAGE = "img";
+    static String DATA = "data";
+    static String NATIVE = "native";
+    static String REQUEST = "request";
+
+
+    public NativeAdUnit(@NonNull String configId) {
+        super(configId, AdType.NATIVE);
+    }
+
+    HashMap<String, Object> requestConfig = new HashMap<>();
+
+    public enum CONTEXT_TYPE {
+        CONTENT_CENTRIC(1),
+        SOCIAL_CENTRIC(2),
+        PRODUCT(3),
+        CUSTOM(500);
+        private int id;
+
+        CONTEXT_TYPE(final int id) {
+            this.id = id;
+        }
+
+        public int getID() {
+            return this.id;
+        }
+
+        public void setID(int id) {
+            if (this.equals(CUSTOM) && id >= 500) {
+                this.id = id;
+            }
+        }
+    }
+
+    public void setContextType(CONTEXT_TYPE type) {
+        requestConfig.put(CONTEXT, type.getID());
+    }
+
+    public enum CONTEXTSUBTYPE {
+        GENERAL(10),
+        ARTICAL(11),
+        VIDEO(12),
+        AUDIO(13),
+        IMAGE(14),
+        USER_GENERATED(15),
+        GENERAL_SOCIAL(20),
+        EMAIL(21),
+        CHAT_IM(22),
+        SELLING(30),
+        APPLICATION_STORE(31),
+        PRODUCT_REVIEW_SITES(32),
+        CUSTOM(500);
+        private int id;
+
+        CONTEXTSUBTYPE(final int id) {
+            this.id = id;
+        }
+
+        public int getID() {
+            return this.id;
+        }
+
+        public void setID(int id) {
+            if (this.equals(CUSTOM) && id >= 500) {
+                this.id = id;
+            }
+        }
+    }
+
+    public void setContextSubType(CONTEXTSUBTYPE type) {
+        requestConfig.put(CONTEXT_SUB_TYPE, type.getID());
+    }
+
+    public enum PLACEMENTTYPE {
```

Comment:
I propose to use `enum PLACEMENT_TYPE` as name of enum and `String PLACEMENT_TYPE_NAME`. The same for `CONTEXTSUBTYPE`.